sender.go

```diff
@@ -186,6 +186,8 @@ func checkMessage(msg *Message) error {
 	} else if msg.TimeToLive < 0 || 2419200 < msg.TimeToLive {
 		return errors.New("the message's TimeToLive field must be an integer " +
 			"between 0 and 2419200 (4 weeks)")
+	} else if msg.Notification == nil {
```

Comment:
Maybe we can check for empty string of both fields and remove pointer

Reply:
That's good too but why we have to remove pointer?

Reply:
I think it is better to follow one approach without mixing pointers and values, we also trying to use values in our backend. I personally prefer this way because it will not allow method to change its argument's value (that is often surprising and error-prone)

Reply:
ah I see. Okie

Reply:
better to have `isValid` method, how do you think?